src/core/common/owning_list.hpp

```diff
@@ -128,6 +125,30 @@ template <typename Type> class OwningList : public LinkedList<Type>
     {
         return OwnedPtr<Type>(LinkedList<Type>::RemoveMatching(aIndicator));
     }
+
+    /**
+     * This template method removes all entries in the list matching a given entry indicator from the list and adds
+     * them to a new list.
+     *
+     * The template type `Indicator` specifies the type of @p aIndicator object which is used to match against entries
+     * in the list. To check that an entry matches the given indicator, the `Matches()` method is invoked on each
+     * `Type` entry in the list. The `Matches()` method should be provided by `Type` class accordingly:
+     *
+     *     bool Type::Matches(const Indicator &aIndicator) const
+     *
+     * @note This method does not change the removed entry itself (which is returned in case of success), i.e., the
+     * entry next pointer stays as before.
+     *
+     * @param[in] aIndicator   An entry indicator to match against entries in the list.
+     * @param[in] aRemovedList The list to add the removed entries to.
+     *
+     * @returns A pointer to the removed matching entry if one could be found, or nullptr if no matching entry is found.
+     *
+     */
+    template <typename Indicator> void RemoveAllMatching(const Indicator &aIndicator, OwningList &aRemovedList)
+    {
+        RemoveAllMatching(aIndicator, aRemovedList);
```

Comment:
May document how ownerships are transferred by `RemoveAllMatching`?

Reply:
Good idea. Thanks. Added in the new push.

Reply:
Thanks.